metaflow/monitor.py

```diff
@@ -2,170 +2,158 @@
 
 from contextlib import contextmanager
 
-from .sidecar import SidecarSubProcess
-from .sidecar_messages import Message, MessageTypes
+from metaflow.sidecar import Message, MessageTypes, Sidecar
 
 COUNTER_TYPE = "COUNTER"
 GAUGE_TYPE = "GAUGE"
-MEASURE_TYPE = "MEASURE"
 TIMER_TYPE = "TIMER"
 
 
-class NullMonitor(object):
-    def __init__(self, *args, **kwargs):
-        pass
+class BaseMonitor(object):
+    TYPE = "nullSidecarMonitor"
 
-    def start(self):
-        pass
+    def __init__(self, flow=None, env=None):
```

Comment:
can this be `(self, *args, **kwargs)`?

Reply:
sure. I can make it more generic (and for the logger).

Reply:
`flow` and `env` are not at all used anywhere in this constructor so it's a bit weird seeing them here.